blocks/filter-list/filter-list.js

```diff
@@ -0,0 +1,185 @@
+import { readBlockConfig } from '../../scripts/lib-franklin.js';
+import ffetch from '../../scripts/ffetch.js';
+
+function buildCell(rowIndex) {
+  const cell = rowIndex ? document.createElement('td') : document.createElement('th');
+  if (!rowIndex) cell.setAttribute('scope', 'col');
+  return cell;
+}
+
+function dataMapLookup(dataMap, value) {
+  const foundValue = dataMap.find((item) => item.key === value);
+  return foundValue ? foundValue.value : value;
+}
+
+function populateTable(data, tableFields, dataMap, detailsUrl, dataSource, year, tbody) {
+  data.forEach((item, i) => {
+    const row = document.createElement('tr');
+    const isEvenRow = i % 2 === 0;
+    row.classList.add(isEvenRow ? 'row2' : 'row1');
+    tableFields.forEach(async (field) => {
+      const isDetailsLink = field.includes('(link)');
+      const rawValue = isDetailsLink ? item[field.replace('(link)', '').trim()] : item[field.trim()];
+      const value = dataMapLookup(dataMap, rawValue);
+      const cell = buildCell(1);
+      if (isDetailsLink) {
+        const a = document.createElement('a');
+        a.href = `${detailsUrl}?key=${item.Pkey}&dataSource=${dataSource}&year=${year}`;
+        a.innerText = value;
+        cell.append(a);
+      } else {
+        cell.innerText = value;
+      }
+      row.append(cell);
+    });
+    tbody.append(row);
+  });
+}
+
+async function loadMore(tableFields, dataMap, detailsUrl, dataSource, year, tbody, loadMoreDiv) {
+  const spreadsheet = `/data-source/${dataSource}/${dataSource}-data.json`;
+  const newData = await ffetch(spreadsheet).sheet(year).slice(20).all();
+
+  populateTable(newData, tableFields, dataMap, detailsUrl, dataSource, year, tbody);
+
+  if (loadMoreDiv) {
+    loadMoreDiv.remove();
+  }
+}
+
+// eslint-disable-next-line max-len
+async function onDropdownChange(block, dropdown, spreadsheet, year, filters, key, detailsUrl, dataSource, tableFields, dataMap) {
+  const tbody = block.querySelector('tbody');
+  tbody.innerHTML = '';
+  const selectedValue = dropdown.value.includes('-- Select ') ? '' : dropdown.value;
+  if (selectedValue === '') {
+    const data = await ffetch(spreadsheet).sheet(year).all();
```

Comment:
I'd generally recommend revisiting calls to `all()` and refactoring to a `for...of` loop. I won't add a comment to every one I see :)